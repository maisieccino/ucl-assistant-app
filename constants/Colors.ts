const black = `#000`
const white = `#fff`

export default {
  accentColor: `#1B998B`,
  accentColorLight: `#73CCC2`,
  black,
  buttonBackground: `#4BB3FD`,
  cardBackground: white,
  cardHeader: `#6E707F`,
  disabledButtonBackground: `rgba(231, 231, 233, 1)`,
  dividerLine: `#808080`,
  errorColor: `rgba(215, 38, 61, 1)`,
  graphCurrentBar: `rgba(27, 152, 137, 0.8)`,
  graphCurrentTime: `rgba(228, 51, 28, 0.65)`,
  indicatorGreen: `#6DD149`,
  indicatorLime: `#A6BC35`,
  indicatorOrange: `#D77026`,
  indicatorRed: `#E4331C`,
  indicatorYellow: `#FFB800`,
  infoColor: `#1C5C8A`,
  lightTextColor: `#6E707F`,
  linkBlue: `#2980b9`,
  oldCardBackground: `#f2f2f2`,
  pageBackground: `#fefefe`,
  tabBackground: `#ced0d9`,
  textColor: `#4D5061`,
  textInputBackground: `#eaeaea`,
  warningColor: `#AB7627`,
  white,
}
