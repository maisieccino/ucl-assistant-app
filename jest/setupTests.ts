import '@testing-library/jest-native/extend-expect'
import '@testing-library/react-native/cleanup-after-each'
import 'react-native-gesture-handler/jestSetup'

// Silence the warning: Animated: `useNativeDriver` is not supported because
// the native animated module is missing
// https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
jest.mock(`react-native/Libraries/Animated/src/NativeAnimatedHelper`)
