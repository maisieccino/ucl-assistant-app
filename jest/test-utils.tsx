import { NavigationContainer } from '@react-navigation/native'
import { render as rtlRender, RenderResult } from '@testing-library/react-native'
import React from 'react'
import { Provider } from 'react-redux'
import { Store } from 'redux'
import configureStore from 'redux-mock-store'
import thunk from "redux-thunk"
import debounce from "../lib/debounce"
import { initialState as defaultInitialState } from '../reducers'

const middlewares = [
  debounce.middleware,
  thunk,
]
const mockStore = configureStore(middlewares)

const render = (
  ui: React.ReactElement,
  {
    initialState = defaultInitialState,
    store = mockStore(initialState) as Store,
    ...renderOptions
  } = {},
): RenderResult => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <NavigationContainer>
        {children}
      </NavigationContainer>
    </Provider>
  )
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

export const waitForEventLoop = async (): Promise<void> => new Promise((resolve) => {
  const done = () => {
    jest.useFakeTimers()
    resolve()
  }
  jest.useRealTimers()
  setTimeout(done, 0)
})

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }
