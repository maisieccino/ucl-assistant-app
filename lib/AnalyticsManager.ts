import * as Amplitude from 'expo-analytics-amplitude'
import Constants from 'expo-constants'

import configureStore from "../configureStore"

const { store } = configureStore

const AMPLITUDE_API_KEY = Constants.manifest.extra
  && Constants.manifest.extra.amplitude
  && Constants.manifest.extra.amplitude.apiKey
  ? Constants.manifest.extra.amplitude.apiKey
  : null

const initialise = (): Promise<void> => Amplitude.initialize(AMPLITUDE_API_KEY)

const shouldTrackAnalytics = (): boolean => (
  AMPLITUDE_API_KEY
  && !__DEV__
  && store.getState().user.settings.shouldTrackAnalytics
)

const setUserId = (userId): void => {
  if (!shouldTrackAnalytics()) { return }

  // make sure this is unique
  if (userId) {
    Amplitude.setUserId(userId)
  }
}

const setUserProperties = ({ ...userProperties }): void => {
  if (!shouldTrackAnalytics()) { return }

  Amplitude.setUserProperties({ ...userProperties })
}

const clearUserProperties = (): void => {
  Amplitude.clearUserProperties()
}

const logEvent = (eventName: string, eventProperties?): void => {
  if (!shouldTrackAnalytics()) { return }

  if (eventProperties) {
    Amplitude.logEventWithProperties(eventName, eventProperties)
  } else {
    Amplitude.logEvent(eventName)
  }
}

const logScreenView = (screenName: string): void => {
  if (!shouldTrackAnalytics()) { return }

  logEvent(`VIEW_SCREEN_${screenName}`)
}

const events = {
  NOTIFICATIONS_ENABLE: `NOTIFICATIONS_ENABLE`,
  NOTIFICATIONS_SKIP: `NOTIFICATIONS_SKIP`,
  PUSH_NOTIFICATIONS_REGISTER: `PUSH_NOTIFICATIONS_REGISTER`,
  SETTINGS_GIVE_FEEDBACK: `SETTINGS_GIVE_FEEDBACK`,
  SETTINGS_RATE_APP: `SETTINGS_RATE_APP`,
  SETTINGS_VIEW_FAQS: `SETTINGS_VIEW_FAQS`,
}

export const AnalyticsManager = {
  clearUserProperties,
  events,
  initialise,
  logEvent,
  logScreenView,
  setUserId,
  setUserProperties,
}

export default AnalyticsManager
