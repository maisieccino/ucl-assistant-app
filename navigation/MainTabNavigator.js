import { Feather } from "@expo/vector-icons"
import { createBottomTabNavigator } from "@react-navigation/bottom-tabs"
import React from "react"

import Colors from "../constants/Colors"
import PeopleScreen from "../screens/PeopleScreen"
import RoomsScreen from "../screens/RoomsScreen"
import SettingsScreen from "../screens/SettingsScreen"
import StudySpacesScreen from "../screens/StudySpacesScreen"
import TimetableScreen from "../screens/TimetableScreen"

const Tab = createBottomTabNavigator()

const screenOptions = ({ route }) => ({
  tabBarIcon: ({ focused }) => {
    let iconName
    switch (route.name) {
      case `Timetable`:
        iconName = `calendar`
        break
      case `StudySpaces`:
        iconName = `book`
        break
      case `People`:
        iconName = `users`
        break
      case `Rooms`:
        iconName = `map-pin`
        break
      case `Settings`:
        iconName = `settings`
        break
      default:
        iconName = `info`
    }
    return (
      <Feather
        name={iconName}
        size={28}
        color={focused ? Colors.pageBackground : Colors.textColor}
      />
    )
  },
})

const tabBarOptions = {
  activeBackgroundColor: Colors.accentColor,
  activeTintColor: Colors.pageBackground,
  allowFontScaling: false,
  bottomNavigationOptions: {
    backgroundColor: Colors.accentColor,
    labelColor: Colors.pageBackground,
    rippleColor: Colors.pageBackground,
  },
  inactiveBackgroundColor: Colors.tabBackground,
  inactiveTintColor: Colors.textColor,
  labelStyle: {
    fontFamily: `apercu`,
  },
  style: {
    backgroundColor: Colors.tabBackground,
    bottom: 0,
    height: 60,
    left: 0,
    position: `absolute`,
    right: 0,
  },
  tabStyle: {
    paddingBottom: 5,
    paddingTop: 5,
  },
}

const MainTabNavigator = () => (
  <Tab.Navigator
    screenOptions={screenOptions}
    initialRouteName="Timetable"
    tabBarOptions={tabBarOptions}
  >
    <Tab.Screen
      name="Timetable"
      component={TimetableScreen}
    />
    <Tab.Screen
      name="StudySpaces"
      component={StudySpacesScreen}
    />
    <Tab.Screen
      name="People"
      component={PeopleScreen}
    />
    <Tab.Screen
      name="Rooms"
      component={RoomsScreen}
    />
    <Tab.Screen
      name="Settings"
      component={SettingsScreen}
    />
  </Tab.Navigator>
)

export default MainTabNavigator
