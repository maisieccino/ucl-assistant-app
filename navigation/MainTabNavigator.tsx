import { Feather } from "@expo/vector-icons"
import { createBottomTabNavigator } from "@react-navigation/bottom-tabs"
import React, { ReactElement } from "react"
import Colors from "../constants/Colors"
import PeopleNavigator from "../screens/People"
import type { PeopleNavigatorParamList } from "../screens/People/PeopleNavigator"
import RoomsNavigator from "../screens/Rooms"
import type { RoomsNavigatorParamList } from "../screens/Rooms/RoomsNavigator"
import SettingsNavigator from "../screens/Settings"
import type { SettingsNavigatorParamList } from "../screens/Settings/SettingsNavigator"
import StudySpacesNavigator from "../screens/StudySpaces"
import type { StudySpacesNavigatorParamList } from "../screens/StudySpaces/StudySpacesNavigator"
import TimetableNavigator from "../screens/Timetable"
import type { TimetableNavigatorParamList } from "../screens/Timetable/TimetableNavigator"
import type { NestedNavigator } from "../types/uclapi"

export type MainTabNavigatorParamList = {
  Timetable: NestedNavigator<TimetableNavigatorParamList>,
  StudySpaces: NestedNavigator<StudySpacesNavigatorParamList>,
  People: NestedNavigator<PeopleNavigatorParamList>,
  Rooms: NestedNavigator<RoomsNavigatorParamList>,
  Settings: NestedNavigator<SettingsNavigatorParamList>,
}

const Tab = createBottomTabNavigator<MainTabNavigatorParamList>()

const screenOptions = ({ route }) => ({
  tabBarIcon: ({ focused }) => {
    let iconName
    switch (route.name) {
      case `Timetable`:
        iconName = `calendar`
        break
      case `StudySpaces`:
        iconName = `book`
        break
      case `People`:
        iconName = `users`
        break
      case `Rooms`:
        iconName = `map-pin`
        break
      case `Settings`:
        iconName = `settings`
        break
      default:
        iconName = `info`
    }
    return (
      <Feather
        name={iconName}
        size={28}
        color={focused ? Colors.pageBackground : Colors.textColor}
      />
    )
  },
})

const tabBarOptions = {
  activeBackgroundColor: Colors.accentColor,
  activeTintColor: Colors.pageBackground,
  allowFontScaling: false,
  bottomNavigationOptions: {
    backgroundColor: Colors.accentColor,
    labelColor: Colors.pageBackground,
    rippleColor: Colors.pageBackground,
  },
  inactiveBackgroundColor: Colors.tabBackground,
  inactiveTintColor: Colors.textColor,
  labelStyle: {
    fontFamily: `apercu`,
  },
  style: {
    backgroundColor: Colors.tabBackground,
    bottom: 0,
    height: 60,
    left: 0,
    right: 0,
  },
  tabStyle: {
    paddingBottom: 5,
    paddingTop: 5,
  },
}

const MainTabNavigator = (): ReactElement => (
  <Tab.Navigator
    screenOptions={screenOptions}
    initialRouteName="Timetable"
    tabBarOptions={tabBarOptions}
  >
    <Tab.Screen
      name="Timetable"
      component={TimetableNavigator}
    />
    <Tab.Screen
      name="StudySpaces"
      component={StudySpacesNavigator}
    />
    <Tab.Screen
      name="People"
      component={PeopleNavigator}
    />
    <Tab.Screen
      name="Rooms"
      component={RoomsNavigator}
    />
    <Tab.Screen
      name="Settings"
      component={SettingsNavigator}
    />
  </Tab.Navigator>
)

export default MainTabNavigator
