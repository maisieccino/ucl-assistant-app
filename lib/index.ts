import AnalyticsManager from './AnalyticsManager'
import ApiManager from './ApiManager'
import AssetManager from './AssetManager'
import debounce from './debounce'
import DeviceManager from './DeviceManager'
import ErrorManager from './ErrorManager'
import LocalisationManager from './LocalisationManager'
import MailManager from './MailManager'
import MapsManager from './MapsManager'
import PushNotificationsManager from './PushNotificationsManager'
import Shadow from './Shadow'
import WebBrowserManager from './WebBrowserManager'

const Random = {
  array: (arr) => arr[Math.floor(Math.random() * arr.length)],
}

export {
  AnalyticsManager,
  ApiManager,
  AssetManager,
  DeviceManager,
  ErrorManager,
  LocalisationManager,
  MailManager,
  MapsManager,
  PushNotificationsManager,
  Random,
  Shadow,
  WebBrowserManager,
  debounce,
}
