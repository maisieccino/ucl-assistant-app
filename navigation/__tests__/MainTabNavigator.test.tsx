import { cleanup, fireEvent } from "@testing-library/react-native"
import React from 'react'
import Colors from "../../constants/Colors"
import { render } from "../../jest/test-utils"
import { Warnings } from '../../lib'
import MainTabNavigator from '../MainTabNavigator'

describe(`MainTabNavigator`, () => {
  beforeAll(() => {
    Warnings.ignore()
    jest.useFakeTimers()
  })

  afterEach(() => {
    cleanup()
  })

  it(`renders without error`, () => {
    const Navigator = render(<MainTabNavigator />)
    expect(Navigator).toMatchSnapshot()
  })

  it(`TimetableScreen is highlighted`, () => {
    const Navigator = render(<MainTabNavigator />)
    const { queryByText } = Navigator
    const timetableTab = queryByText(`Timetable`)
    expect(timetableTab).toBeTruthy();
    (expect(timetableTab) as any).toHaveStyle({ color: Colors.pageBackground })
    const TimetableScreen = queryByText(`Loading timetable...`)
    expect(TimetableScreen).toBeTruthy()
  })

  it(`can navigate to SettingsScreen`, () => {
    const Navigator = render(<MainTabNavigator />)
    const { queryByText } = Navigator
    const settingsTab = queryByText(`Settings`)
    expect(settingsTab).toBeTruthy();
    (expect(settingsTab) as any).toHaveStyle({ color: Colors.textColor })

    fireEvent.press(settingsTab);
    (expect(settingsTab) as any).toHaveStyle({ color: Colors.pageBackground })

    const signOutButton = queryByText(`Sign Out`)
    expect(signOutButton).toBeTruthy()
  })
})
