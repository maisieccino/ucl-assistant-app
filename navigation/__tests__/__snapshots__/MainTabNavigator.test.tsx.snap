// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTabNavigator renders without error 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            null,
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                onLayout={[Function]}
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    accessibilityElementsHidden={false}
                    closing={false}
                    gestureVelocityImpact={0.3}
                    importantForAccessibility="auto"
                    onClose={[Function]}
                    onGestureBegin={[Function]}
                    onGestureCanceled={[Function]}
                    onOpen={[Function]}
                    onTransitionStart={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                    transitionSpec={
                      Object {
                        "close": Object {
                          "animation": "spring",
                          "config": Object {
                            "damping": 500,
                            "mass": 3,
                            "overshootClamping": true,
                            "restDisplacementThreshold": 10,
                            "restSpeedThreshold": 10,
                            "stiffness": 1000,
                          },
                        },
                        "open": Object {
                          "animation": "spring",
                          "config": Object {
                            "damping": 500,
                            "mass": 3,
                            "overshootClamping": true,
                            "restDisplacementThreshold": 10,
                            "restSpeedThreshold": 10,
                            "stiffness": 1000,
                          },
                        },
                      }
                    }
                  >
                    <View
                      pointerEvents="box-none"
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        onGestureHandlerEvent={[Function]}
                        onGestureHandlerStateChange={[Function]}
                        style={
                          Object {
                            "flex": 1,
                            "transform": Array [
                              Object {
                                "translateX": 0,
                              },
                              Object {
                                "translateX": 0,
                              },
                            ],
                          }
                        }
                      >
                        <View
                          pointerEvents="none"
                          style={
                            Object {
                              "backgroundColor": "rgb(242, 242, 242)",
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "shadowColor": "#000",
                              "shadowOffset": Object {
                                "height": 1,
                                "width": -1,
                              },
                              "shadowOpacity": 0.3,
                              "shadowRadius": 5,
                              "top": 0,
                              "width": 3,
                            }
                          }
                        />
                        <View
                          style={
                            Array [
                              Object {
                                "flex": 1,
                                "overflow": "hidden",
                              },
                              Array [
                                Object {
                                  "backgroundColor": "rgb(242, 242, 242)",
                                },
                                undefined,
                              ],
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "flexDirection": "column-reverse",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "flex": 1,
                                }
                              }
                            >
                              <RCTSafeAreaView
                                emulateUnlessSupported={true}
                                style={
                                  Object {
                                    "backgroundColor": "#fefefe",
                                    "flex": 1,
                                    "flexDirection": "column",
                                    "paddingBottom": 60,
                                  }
                                }
                              >
                                <View
                                  onLayout={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "backgroundColor": "#fefefe",
                                        "flex": 1,
                                        "flexDirection": "column",
                                        "paddingBottom": 60,
                                      },
                                      Object {
                                        "paddingBottom": 0,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "#fefefe",
                                          "flex": 1,
                                          "paddingLeft": 20,
                                          "paddingRight": 20,
                                        },
                                        Object {
                                          "padding": 20,
                                        },
                                      ]
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "flex": 1,
                                          "justifyContent": "center",
                                        }
                                      }
                                    >
                                      <Text
                                        style={
                                          Array [
                                            Object {
                                              "color": "#4D5061",
                                              "fontFamily": "apercu",
                                              "fontSize": 16,
                                              "marginBottom": 1,
                                              "marginTop": 1,
                                            },
                                            Object {
                                              "marginBottom": 20,
                                            },
                                          ]
                                        }
                                      >
                                        Loading timetable...
                                      </Text>
                                      <ActivityIndicator
                                        animating={true}
                                        color="#999999"
                                        hidesWhenStopped={true}
                                        size="large"
                                      />
                                    </View>
                                  </View>
                                </View>
                              </RCTSafeAreaView>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  importantForAccessibility="auto"
                  onLayout={[Function]}
                  pointerEvents="box-none"
                  style={null}
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "#ced0d9",
          "borderTopColor": "rgb(224, 224, 224)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "height": 60,
          "left": 0,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityLabel="Timetable, tab, 1 of 5"
          accessibilityRole="button"
          accessibilityStates={
            Array [
              "selected",
            ]
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "#1B998B",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              Object {
                "paddingBottom": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text />
            </View>
          </View>
          <Text
            allowFontScaling={false}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                Object {
                  "color": "#fefefe",
                },
                Object {
                  "fontSize": 11,
                  "marginBottom": 1.5,
                },
                Object {
                  "fontFamily": "apercu",
                },
              ]
            }
          >
            Timetable
          </Text>
        </View>
        <View
          accessibilityLabel="StudySpaces, tab, 2 of 5"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "#ced0d9",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              Object {
                "paddingBottom": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text />
            </View>
          </View>
          <Text
            allowFontScaling={false}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                Object {
                  "color": "#4D5061",
                },
                Object {
                  "fontSize": 11,
                  "marginBottom": 1.5,
                },
                Object {
                  "fontFamily": "apercu",
                },
              ]
            }
          >
            StudySpaces
          </Text>
        </View>
        <View
          accessibilityLabel="People, tab, 3 of 5"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "#ced0d9",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              Object {
                "paddingBottom": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text />
            </View>
          </View>
          <Text
            allowFontScaling={false}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                Object {
                  "color": "#4D5061",
                },
                Object {
                  "fontSize": 11,
                  "marginBottom": 1.5,
                },
                Object {
                  "fontFamily": "apercu",
                },
              ]
            }
          >
            People
          </Text>
        </View>
        <View
          accessibilityLabel="Rooms, tab, 4 of 5"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "#ced0d9",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              Object {
                "paddingBottom": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text />
            </View>
          </View>
          <Text
            allowFontScaling={false}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                Object {
                  "color": "#4D5061",
                },
                Object {
                  "fontSize": 11,
                  "marginBottom": 1.5,
                },
                Object {
                  "fontFamily": "apercu",
                },
              ]
            }
          >
            Rooms
          </Text>
        </View>
        <View
          accessibilityLabel="Settings, tab, 5 of 5"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "#ced0d9",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              Object {
                "paddingBottom": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text />
            </View>
          </View>
          <Text
            allowFontScaling={false}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                Object {
                  "color": "#4D5061",
                },
                Object {
                  "fontSize": 11,
                  "marginBottom": 1.5,
                },
                Object {
                  "fontFamily": "apercu",
                },
              ]
            }
          >
            Settings
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
